lib/stm32/common/i2c_common_v1.c

```diff
@@ -65,6 +65,11 @@ void i2c_reset(uint32_t i2c)
 		rcc_periph_reset_pulse(RST_I2C3);
 		break;
 #endif
+#if defined(I2C4_BASE)
+	case I2C4:
+		rcc_periph_reset_pulse(RCC_I2C4);
+		break;
```

Comment:
This reminds me once again of the enum problem. @karlp suggested you add `rcc_periph_reset_pulse(RCC_xxx)` but the actual enum constant **should be** RST_xxx which is shown above in line 65. This confusion between RCC_periph and RST_periph is a ripe source of errors in the code base.